main.tf

```diff
@@ -43,13 +45,29 @@ resource "mongodbatlas_project_ip_access_list" "additional_cidr" {
   cidr_block = var.override_peering_cidr
 }
 
+# To be removed on next MAJOR release
 resource "mongodbatlas_project_ip_access_list" "public_ips" {
   count = var.create_vpc_peering || var.create_privatelink ? 0 : length(var.vpc_public_ips)
 
   project_id = local.project_id
   ip_address = var.vpc_public_ips[count.index]
 }
 
+resource "mongodbatlas_project_ip_access_list" "ips" {
+  for_each = { for k, v in var.ip_access_list : v.ip => v }
+
+  project_id = local.project_id
+  ip_address = replace(each.value.ip, "/32", "")
```

Comment:
I wouldnt do any magic here and specify that it should be a ip address not not a cidr.